src/expr.h

```diff
@@ -82,6 +82,10 @@ class Expr : public ASTNode {
     virtual Expr *Instantiate(TemplateInstantiation &templInst) const = 0;
 
     virtual bool HasAmbiguousVariability(std::vector<const Expr *> &warn) const;
+
+    /** This method is used to check expressions on signed operations
+        for overflow attributes. Returns false by default. */
+    virtual bool IsSigned() const;
```

Comment:
Fix description

Reply:
Why do we need a function for that, it's basically the check that `Expr::GetType()` is a signed integer type, that's it. This way you don't need to check it differently for different types on expressions. And you can do the check of this property locally, instead of adding new function.

Reply:
By definition, the type of operation is ruling if the `nsw` attribute needs to be applied or not. `IsSigned()` method is really a hack to achieve the same result though the duplicating logic. If using the type of operation contradicts the logic currently implemented in `IsSigned()` - this needs to be investigated and fixed.

Generally speaking, 99% of logic in the front-end is implementation of the rules in the language standard. And the language standard is pretty clear here. There are rules for defining operation type (it's a combination of promotion rules and rule that find the common type for the binary operations) and operation type is governing the properties of the computation (`nsw` attribute). So we should follow this logic instead of trying to hack it around and implement the shortcuts, which actually duplicating existing logic.